plugins/scaffolder-backend/src/lib/repo/disk.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import glob from 'glob';
+import * as fs from 'fs-extra';
+import { logger } from 'lib/logger';
+import { Template, RepositoryBase as Base } from '.';
+
+interface DiskIndexEntry {
+  contents: Template;
+  location: string;
+}
+
+export class DiskRepository implements Base {
+  private repository: Template[] = [];
+  private localIndex: DiskIndexEntry[] = [];
+
+  constructor(private repoDir = `${__dirname}/templates`) {
+    this.reindex();
+  }
+
+  public async list(): Promise<Template[]> {
+    return this.repository;
+  }
+
+  public async reindex(): Promise<void> {
+    this.localIndex = await this.index();
+    this.repository = this.localIndex.map(({ contents }) => contents);
+  }
+
+  public async prepare(templateId: string): Promise<string> {
+    const template = this.localIndex.find(
+      ({ contents }) => contents.id === templateId,
+    );
+
+    if (!template) {
+      throw new Error('Template no found');
+    }
+
+    const tempDir = await fs.promises.mkdtemp(templateId);
+    await fs.copy(template.location, tempDir);
+    return tempDir;
+  }
+
+  private async index(): Promise<DiskIndexEntry[]> {
+    return new Promise((resolve, reject) => {
+      glob(`${this.repoDir}/**/template-info.json`, async (err, matches) => {
+        if (err) {
+          reject(err);
+        }
+
+        const fileContents: Array<{
+          location: string;
+          contents: string;
+        }> = await Promise.all(
+          matches.map(async (location: string) => ({
+            location,
+            contents: await fs.readFile(location, 'utf-8'),
+          })),
+        );
+
+        const validFiles = fileContents.reduce(
```

Comment:
Yeah I guess so, I guess I just liked the idea that it would reduce away the problematic templates ;) haha.

But for readability maybe it makes more sense